propolis/src/instance.rs

```diff
@@ -221,6 +221,11 @@ impl Instance {
         func(machine, &mctx, &self.disp, &state.inv)
     }
 
+    pub fn machine(&self) -> Arc<Machine> {
+        let state = self.inner.lock().unwrap();
+        state.machine.clone().unwrap()
+    }
```

Comment:
This is fine but as part of trying plumbing a migrate state to propolis and tie in the migrate tasks, I just gave them `AsyncCtx` parameters which allows grabbing the `DispCtx` which has a `MachineCtx` that just wraps a `Machine`.

https://github.com/oxidecomputer/propolis/blob/6a8a3ad4d09e0b7f225eba375115c41ddacb1f4e/server/src/lib/migrate/destination.rs#L18-L30

Reply:
That seems better than emitting the `Machine` straight out.

Reply:
Agreed. That's better in all ways.